client/components/webpack-build-monitor/index.jsx

```diff
@@ -5,147 +5,113 @@
  * External dependencies
  */
 import React from 'react';
-import { createStore } from 'redux';
 import classNames from 'classnames';
-import { find, includes, startsWith, identity } from 'lodash';
+import { find, startsWith } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Spinner from 'components/spinner';
-import { combineReducers } from 'state/utils';
-
-// Action dispatched by wrapped console
-const CONSOLE_MESSAGE = 'CONSOLE_MESSAGE';
-
-const CONNECTED = 'CONNECTED';
-const DISCONNECTED = 'DISCONNECTED';
-const BUILDING_CSS = 'BUILDING_CSS';
-const BUILDING_JS = 'BUILDING_JS';
-const BUILDING_BOTH = 'BUILDING_BOTH';
-const ERROR = 'ERROR';
-const IDLE = 'IDLE';
-
-const doneBuilding = typeDone => state => {
-	if ( state === ERROR ) {
-		return state;
-	} else if ( state === BUILDING_BOTH ) {
-		return typeDone === BUILDING_JS ? BUILDING_CSS : BUILDING_JS;
-	}
-	return IDLE;
-};
-
-const buildJs = state => ( state === BUILDING_CSS ? BUILDING_BOTH : BUILDING_JS );
-const buildCss = state => ( state === BUILDING_JS ? BUILDING_BOTH : BUILDING_CSS );
 
 const MESSAGE_STATUS_MAP = {
-	'[HMR] connected': () => CONNECTED,
-	'[WDS] Disconnected!': () => DISCONNECTED,
-	'[HMR] bundle rebuilding': state => buildJs( state ),
-	'Building CSS…': state => buildCss( state ),
-	'CSS build failed': () => ERROR,
-	'[WDS] Nothing changed.': doneBuilding( BUILDING_JS ),
-	'Reloading CSS: ': doneBuilding( BUILDING_CSS ),
-	'[HMR] bundle rebuilt in': doneBuilding( BUILDING_JS ),
+	'[HMR] connected': { isConnected: true },
+	'[WDS] Disconnected!': { isConnected: false },
+	'[HMR] bundle rebuilding': { isBuildingJs: true },
+	"[HMR] The following modules couldn't be hot updated": { needsReload: true },
```

Comment:
I just came across another string that should correspond to `{ needsReload: true }`:

```
[HMR] Cannot find update (Full reload needed)
```

This happens if you kill the build and restart.

Reply:
Trapped in 53a6b7da1